### PR TITLE
Increase test coverage in `pkg/util`

### LINKED ISSUE #3871: Increase unit test coverage in `pkg/util`
### Proposal

https://github.com/kedacore/keda/tree/main/pkg/util

1) Following files don't have unit test, we should target in having majority of those covered
- `gvkr.go`
- `http.go`
- `normalize_string.go`
- `tls_config.go`

2) We should inspect the rest of the package and increase test coverage if needed

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dttung2905**
Hi team,
This PR will increase the test coverage in `pkg/util` 

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #3871

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ Here is an overview of all new **experimental** features:
 
 - **General**: Add a warning when KEDA run outside supported k8s versions ([#4130](https://github.com/kedacore/keda/issues/4130))
 - **General**: Use (self-signed) certificates for all the communications (internals and externals) ([#3931](https://github.com/kedacore/keda/issues/3931))
+- **General**: Improve test coverage in `pkg/util` ([#3871](https://github.com/kedacore/keda/issues/3871))
```
Comment (zroubalik) [inline CHANGELOG.md:65]: just a nit, please move this to `### Other` sections as it is not user facing feature

Comment (zroubalik) [reply CHANGELOG.md:65]: and pls don't forget to sort alphabetically

**Review by zroubalik — COMMENTED** [review]
Thanks a lot for this!

File: CHANGELOG.md (hunk 1 above)
Comment (dttung2905) [reply CHANGELOG.md]: Thanks for the review. I have move that line to `Other` and did some minor formatting too :pray:

**Review by zroubalik — APPROVED** [review]
LGTM,

excellent, thanks a lot for the contribution!

**Pull request merged**